docs/changelog-entries/262.md

```diff
@@ -0,0 +1 @@
+- Merged `initializeData` into `initialize` [#262](https://github.com/precice/openfoam-adapter/pull/262).
```

Comment:
Let's update this to something like "Migrated the adapter to the preCICE v3 API" and also update the PR title and description, to include all related changes.